compile-release-tools

```diff
@@ -71,6 +70,8 @@ main() {
       echo "Compiling default release tools..."
       compile "k8s.io/release/cmd/${tool}"
     done
+    # krel will be built via the Makefile
+    make krel krel-install
```

Comment:
Rebased on top of the latest master branch. I would do it this way, WDYT?

Reply:
@saschagrunert -- I think I'd prefer to see the logic that you have in the Makefile in `compile-release-tools` instead.

Ideally, we have just one entrypoint for the release tools which handles build/installation.

Also, pattern-wise, maybe we can have something like this:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/24b2057904692380d419db6e8ab56ec35779993c/version/version.go#L53-L66

i.e., have something like `pkg/version` which can then be used by a `version` subcommand in any release tool.
Let me know what you think!

Reply:
Yes this sounds good, I changed the implementation to contain a dedicated `version` package in `pkg`. The I adapted the `compile-release-tools` to contain the necessary changes. PTAL